### DISCUSSION TITLE
Unable to migrate from v2.6 to v2.7

### DISCUSSION THREAD

**Discussion by Jaarson** [formbricks/formbricks#4357] [Self-hosting] [answered]
### Issue Summary

I use self-hosted version. Currently on v.2.6.0. When I run the database migration command (migration to v2.7) it fails with an error and Formbricks v2.7.0 can't launch.

I'm using `pgvector/pgvector:pg15` image.

When I run migration command for v2.6 (my current version) it's OK, says `Data migration completed`

### Expected Behavior

Migration should be successful

### Other information (incl. screenshots, Formbricks version, steps to reproduce,...)

Error from `formbricks-migration` container
```
 > @formbricks/database@0.1.0 data-migration:v2.7 /app/packages/database
 > pnpm data-migration:add-teams


 > @formbricks/database@0.1.0 data-migration:add-teams /app/packages/database
 > ts-node ./data-migrations/20241107161932_add_teams/data-migration.ts

 Starting data migration...
 An error occurred during migration: PrismaClientKnownRequestError:
 Invalid `transactionPrisma.invite.findMany()` invocation in
 /app/packages/database/data-migrations/20241107161932_add_teams/data-migration.ts:26:55

   23 await prisma.$transaction(
   24   async (transactionPrisma) => {
   25     // Fetch all invites and group them by organizationId
 → 26     const invites = (await transactionPrisma.invite.findMany(
 The column `Invite.deprecatedRole` does not exist in the current database.
     at Mn.handleRequestError (/app/node_modules/@prisma/client/runtime/library.js:121:7753)
     at Mn.handleAndLogRequestError (/app/node_modules/@prisma/client/runtime/library.js:121:7061)
     at Mn.request (/app/node_modules/@prisma/client/runtime/library.js:121:6745)
     at async l (/app/node_modules/@prisma/client/runtime/library.js:130:9633)
     at async prisma.$transaction.timeout (/app/packages/database/data-migrations/20241107161932_add_teams/data-migration.ts:26:24)
     at async Proxy._transactionWithCallback (/app/node_modules/@prisma/client/runtime/library.js:130:8000)
     at async runMigration (/app/packages/database/data-migrations/20241107161932_add_teams/data-migration.ts:23:3) {
   code: 'P2022',
   clientVersion: '5.20.0',
   meta: { modelName: 'Invite', column: 'Invite.deprecatedRole' }
 }
  ELIFECYCLE  Command failed with exit code 1.
  ELIFECYCLE  Command failed with exit code 1.
```

### Your Environment

Self-hosted Formbricks

**Comment by saschafoerster** [comment]
It's the same as #4294, can be closed as duplicate, I guess. But sad to hear, that you have the same issue, @Jaarson.

**Comment by mattinannt** [comment]
@Jaarson Did you start the docker container with the new Formbricks 2.7 version before running the migration?
It seems like your database schema is still on the 2.6 version which should auto-migrate, once Formbricks 2.7 container has been started successfully. Can you please also share the logs from the formbricks docker container?

**Comment by Jaarson** [comment]
@mattinannt initially yes, I ran a 2.7 image by accident, but luckily it was all running locally. Then I restored Postgre from a backup and went through the migration process as described in the docs, still running 2.6. However, the effect was the same.

**Comment by mattinannt** [comment, marked as answer]
@Jaarson If the database schema migration failed in the Formbricks container setup, this should solve the issue:
https://github.com/formbricks/formbricks/issues/4294#issuecomment-2498110701

**Comment by Jaarson** [comment]
Thanks @mattinannt 💪 I'm looking forward to automatic migrations. A nice example is Metabase. Just pull the latest image, run it, and it just works. Of course, I totally understand that updating Postgres to use pgvector is not easy (or at all) possible using auto-updates. Anyway, I absolutely love Formcricks!
